src/app/access-requests/services/access-request.ts

```diff
@@ -424,20 +424,34 @@ export class AccessRequestService {
   });
 
   /**
-   * This function attempts to revoke an access grant in the backend.
-   * @param grantID the ID of the grant to revoke.
-   * @returns a Promise that evaluates to true if successful, false otherwise.
+   * Remove the grant locally.
+   * @param id - the ID of the grant to remove
    */
-  revokeAccessGrant(grantID: string): Promise<boolean> {
-    if (!grantID || grantID === '') {
-      return Promise.resolve(false);
+  #removeGrantLocally(id: string): void {
```

Comment:
Something doesn't look right here. This method should remove a grant in the cache, but instead it seems to update a grant.

The effect should be that the grant should not show up in the UI any more, even with mock backend.

Apropos mock backend: I think the URL for deletion in the mock backend is wrong, so you always get an error when deleting a grant.